guardrails/src/drivers/rest/routers/admin.py

```diff
@@ -0,0 +1,273 @@
+) -> ConfigResponse:
+    """Get a guardrail configuration by ID."""
+    config = await uc.execute(config_id)
+    return ConfigResponse(
+        id=config.id,
+        domain_id=config.domain_id,
+        name=config.name,
+        description=config.description,
+        config_yaml=config.config_yaml,
+        prompts_yaml=config.prompts_yaml,
+        colang=config.colang,
+        created_at=config.created_at,
+        updated_at=config.updated_at,
+    )
+
+
+@router.put("/{domain_id}/configs/{config_id}", response_model=ConfigResponse, tags=["configs"])
+async def update_config(
+    domain_id: UUID,
+    config_id: UUID,
+    data: ConfigUpdate,
+    uc: UpdateConfig = Depends(get_update_config),
+) -> ConfigResponse:
+    """Update a guardrail configuration."""
+    config = await uc.execute(
+        config_id=config_id,
+        name=data.name,
+        config_yaml=data.config_yaml,
+        prompts_yaml=data.prompts_yaml,
+        colang=data.colang,
+        description=data.description,
+    )
+    return ConfigResponse(
+        id=config.id,
+        domain_id=config.domain_id,
+        name=config.name,
+        description=config.description,
+        config_yaml=config.config_yaml,
+        prompts_yaml=config.prompts_yaml,
+        colang=config.colang,
+        created_at=config.created_at,
+        updated_at=config.updated_at,
+    )
+
+
+@router.delete("/{domain_id}/configs/{config_id}", tags=["configs"])
+async def delete_config(
+    domain_id: UUID,
+    config_id: UUID,
+    uc: DeleteConfig = Depends(get_delete_config),
+) -> dict:
+    """Delete a guardrail configuration."""
+    await uc.execute(config_id)
+    return {"status": "deleted", "config_id": str(config_id)}
+
+
+# ==================== Version Management ====================
+
+
+@router.post(
+    "/{domain_id}/configs/{config_id}/versions",
+    response_model=VersionResponse,
+    tags=["versions"],
+)
+async def create_version(
+    domain_id: UUID,
+    config_id: UUID,
+    data: VersionCreate,
+    uc: CreateVersion = Depends(get_create_version),
+) -> VersionResponse:
+    """Create a new version for a configuration."""
+    version = await uc.execute(
+        config_id=config_id,
+        name=data.name,
+        description=data.description,
+    )
+    return VersionResponse(
+        id=version.id,
+        config_id=version.config_id,
+        name=version.name,
+        revision=version.revision,
+        is_active=version.is_active,
+        description=version.description,
+        created_at=version.created_at,
+    )
+
+
+@router.get(
+    "/{domain_id}/configs/{config_id}/versions",
+    response_model=List[VersionResponse],
+    tags=["versions"],
+)
+async def list_versions(
+    domain_id: UUID,
+    config_id: UUID,
+    offset: int = Query(0, ge=0),
+    limit: int = Query(100, ge=1, le=1000),
+) -> List[VersionResponse]:
+    """List all versions for a configuration."""
+    repo = get_repository()
```

Comment:
direct call than use case as in other endpoints here, it is inconsistent unlike what is done on lust_configs